examples/src/main/java/SolidityPrecompileExample.java

```diff
@@ -0,0 +1,126 @@
+            AccountId aliceAccountId = Objects.requireNonNull(new AccountCreateTransaction()
+                .setKey(alicePublicKey)
+                .setInitialBalance(Hbar.fromTinybars(1000))
+                .execute(client)
+                .getReceipt(client)
+                .accountId
+            );
+
+            // Instantiate ContractHelper
+
+            ContractHelper contractHelper = new ContractHelper(
+                ContractHelper.getJsonResource("precompile-example/PrecompileExample.json"),
+                new ContractFunctionParameters()
+                    .addAddress(OPERATOR_ID.toSolidityAddress())
+                    .addAddress(aliceAccountId.toSolidityAddress()),
+                client
+            );
+
+            // Configure steps in ContracHelper
+
+            contractHelper
+                .setResultValidatorForStep(0, contractFunctionResult -> {
+                    System.out.println("getPseudoRandomSeed() returned " + Arrays.toString(contractFunctionResult.getBytes32(0)));
+                    return true;
+                }).setPayableAmountForStep(1, Hbar.from(20))
+                // step 3 associates Alice with the token, which requires Alice's signature
+                .addSignerForStep(3, alicePrivateKey)
+                .addSignerForStep(5, alicePrivateKey)
+                .setParameterSupplierForStep(11, () -> {
+                    return new ContractFunctionParameters()
+                        // when contracts work with a public key, they handle the raw bytes of the public key
+                        .addBytes(alicePublicKey.toBytesRaw());
+                }).setPayableAmountForStep(11, Hbar.from(40))
+                // Because we're setting the adminKey for the created NFT token to Alice's key,
+                // Alice must sign the ContractExecuteTransaction.
+                .addSignerForStep(11, alicePrivateKey)
+                // and Alice must sign for minting because her key is the supply key.
+                .addSignerForStep(12, alicePrivateKey)
+                .setParameterSupplierForStep(12, () -> {
+                    return new ContractFunctionParameters()
+                        // add three metadatas
+                        .addBytesArray(new byte[][]{new byte[]{0x01b}, new byte[]{0x02b}, new byte[]{0x03b}});
+                }) // and alice must sign to become associated with the token.
+                .addSignerForStep(13, alicePrivateKey)
+                // Alice must sign to burn the token because her key is the supply key
+                .addSignerForStep(16, alicePrivateKey);
+
+            // step 0 tests pseudo random number generator (PRNG)
+            // step 1 creates a fungible token
+            // step 2 mints it
+            // step 3 associates Alice with it
+            // step 4 transfers it to Alice.
+            // step 5 approves an allowance of the fungible token with operator as the owner and Alice as the spender [NOT WORKING]
```

Comment:
Step 5 function is failing because the created fungible token is associated only with the spender (aliseAccount), but it needs to be also associated with the provider of allowance, which in this case is the calling contract.

Reply:
I would have thought the provider of the allowance would be whoever's ID is in the transaction ID of the `ContractExecuteTransaction`.  Does this mean that the owner of an allowance can only ever be the contract itself?  Is there any way for a contract function to approve an allowance with a different owner?

Reply:
The allowance provider is the `sender` in the message frame when we reach the precompiled contract. 
For example, if when have this setup:
User A --calls-> Contract C --calls-> Precompile, the `sender` would be Contract C.
> Does this mean that the owner of an allowance can only ever be the contract itself?

No, we could also have this setup (notice the delegatecall):
User A --calls-> Contract C --delegatecalls-> Precompile, then the `sender` would be User A.
We have such a contract and test in the services repo in [ApproveAllowanceSuite](https://github.com/hashgraph/hedera-services/blob/bb302309e61eb0b0fc11de967266433951b6bb31/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ApproveAllowanceSuite.java#L209) which uses the [HtsApproveAllowance.sol](https://github.com/hashgraph/hedera-services/blob/bb302309e61eb0b0fc11de967266433951b6bb31/test-clients/src/main/resource/contract/contracts/HtsApproveAllowance/HtsApproveAllowance.sol#L18) contract.